Sources/SourceKitLSP/Swift/ClosureCompletionFormat.swift

```diff
@@ -0,0 +1,63 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftBasicFormat
+
+/// A specialization of `BasicFormat` for closure literals in a code completion
+/// context.
+///
+/// This is more conservative about newline insertion: unless the closure has
+/// multiple statements in its body it will not be reformatted to multiple
+/// lines.
+class ClosureCompletionFormat: BasicFormat {
+  override func requiresNewline(between first: TokenSyntax?, and second: TokenSyntax?) -> Bool {
+    if let first, isEndOfSmallClosureSignature(first) {
+      return false
+    } else if let first, isSmallClosureDelimiter(first, kind: \.leftBrace) {
+      return false
+    } else if let second, isSmallClosureDelimiter(second, kind: \.rightBrace) {
+      return false
+    } else {
+      return super.requiresNewline(between: first, and: second)
+    }
+  }
+
+  /// Returns `true` if `token` is an opening or closing brace (according to
+  /// `kind`) of a closure, and that closure has no more than one statement in
+  /// its body.
+  private func isSmallClosureDelimiter(
+    _ token: TokenSyntax,
+    kind: KeyPath<ClosureExprSyntax, TokenSyntax>
```

Comment:
Moving https://github.com/swiftlang/swift-syntax/pull/2897#discussion_r1847456663 here

> Do we need to pass in the key path here? It doesn’t really look very dynamic, so I’d just check for `\.leftBrace` and `\.rightBrace` in here.

It is defensive, but I think it's worthwhile. We don't want to override the `BasicFormat` for the token _before_ a `{` or _after_ a `}`. Passing the keypath guards against those.

Reply:
Sorry, looks like my comment was misleading. I meant that instead of taking a key path as a parameter to this function, we could hardcode it inside the body (ie. changing the body to `token.keyPathInParent == \.leftBrace || token.keyPathInParent == \.rightBrace `). But upon second thought, that would change semantics because it would also suppress the newline after the right brace. So, you’re implementation is good. Ignore my comment.